src/ert/_c_wrappers/enkf/config/gen_kw_config.py

```diff
@@ -71,53 +32,57 @@ def __init__(
         if not os.path.isfile(parameter_file):
             raise IOError(f"No such file:{parameter_file}")
 
-        c_ptr = self._alloc_empty(key, tag_fmt)
-        if c_ptr:
-            super().__init__(c_ptr)
-        else:
-            raise ValueError(
-                "Could not instantiate GenKwConfig with "
-                f'key="{key}" and tag_fmt="{tag_fmt}"'
-            )
-        self._set_parameter_file(parameter_file)
-        self._set_template_file(template_file)
+        self._key = key
+        self.name = key
+        self._parameter_file = parameter_file
+        self._template_file = template_file
+        self._tag_format = tag_fmt
+        self._transfer_functions = []
+        self.forward_init = ""
+        self.output_file = ""
+        self.forward_init_file = ""
+
+        with open(parameter_file, "r", encoding="utf-8") as file:
+            for item in file:
+                item = item.rsplit("--")[0]  # remove comments
+
+                if item.strip():  # only lines with content
+                    self._transfer_functions.append(self.parse_transfer_function(item))
+
         self.__str__ = self.__repr__
 
-    def getTemplateFile(self) -> os.PathLike[str]:
-        path = self._get_template_file()
+    def getTemplateFile(self) -> str:
+        path = self._template_file
         return None if path is None else os.path.abspath(path)
 
-    def getParameterFile(self):
-        path = self._get_parameter_file()
+    def getParameterFile(self) -> str:
+        path = self._parameter_file
         return None if path is None else os.path.abspath(path)
 
-    def getKeyWords(self) -> StringList:
-        return self._alloc_name_list()
-
-    def shouldUseLogScale(self, index: int) -> bool:
-        return self._should_use_log_scale(index)
-
-    def free(self):
-        self._free()
+    def shouldUseLogScale(self, keyword: str) -> bool:
+        for tf in self._transfer_functions:
+            if tf.name == keyword:
+                return tf._use_log
+        return False
 
     def __repr__(self):
-        return (
-            f'GenKwConfig(key = "{self.getKey()}", '
-            f'tag_fmt = "{self.tag_fmt}") at 0x{self._address():x}'
-        )
+        return f'GenKwConfig(key = "{self.getKey()}", ' f'tag_fmt = "{self.tag_fmt}")'
 
     def getKey(self) -> str:
-        return self._get_key()
+        return self._key
+
+    def getImplementationType(self) -> str:
```

Comment:
Seems like the tests don't call `gen_kw_config.getImplementationType`.

Reply:
This should be covered now. :+1: